### PR TITLE
Fixes #26639 - Experimental Host Wizard

### PR DESCRIPTION
**Pull request by Ron-Lavi**
creating an experimental page where all the fun would happen.
go to: `/host_wizard` page, enjoy the current loading wizard :)

Selection_051

### CODE REVIEW

**Comment by Ron-Lavi** [comment]
@theforeman/ui-ux @tbrisker @ShimShtein could you take a look ?

**Comment by xprazak2** [comment]
Nice start, maybe this could be added to the lab features menu.

**Comment by ohadlevy** [comment]
I would also prefer to wait merging this until we have the react router merged, this will avoid the need for extra controllers etc.

**Comment by Ron-Lavi** [comment]
blocked by #6522

**Review by ohadlevy — CHANGES_REQUESTED** [review]
Once the react router is in core, I expect to see no rails controllers at all in this PR, perhaps you can start it already?

**Comment by waldenraines** [comment]
> Nice start, maybe this could be added to the lab features menu.

Yes, it should be on the labs features menu.

**Comment by Ron-Lavi** [comment]
@glekner @ohadlevy can you help on this? What else do I need here to connect to react-router? thanks :)

**Comment by ohadlevy** [comment]
```patch
diff --git a/app/controllers/react_controller.rb b/app/controllers/react_controller.rb
index 01c733705..7d3d09591 100644
--- a/app/controllers/react_controller.rb
+++ b/app/controllers/react_controller.rb
@@ -1,3 +1,6 @@
 class ReactController < ApplicationController
   layout 'layouts/react_application'
+
+  def index
+  end
 end
diff --git a/app/views/react/index.html.erb b/app/views/react/index.html.erb
new file mode 100644
index 000000000..e69de29bb
diff --git a/config/routes.rb b/config/routes.rb
index 58d197a30..e380cc2f2 100644
--- a/config/routes.rb
+++ b/config/routes.rb
@@ -553,4 +553,7 @@ Foreman::Application.routes.draw do
   if Rails.env.development? && defined?(::GraphiQL::Rails::Engine)
     mount GraphiQL::Rails::Engine, at: '/graphiql', graphql_path: '/api/graphql'
   end
+
+  match 'host_wizard' => 'react#index', :via => :get
+  match 'host_wizard/*id' => 'react#index', :via => :get
 end
```

File: webpack/assets/javascripts/react_app/components/componentRegistry.js
Comment (glekner) [inline webpack/assets/javascripts/react_app/components/componentRegistry.js]: there is no need for the page to be in the registry, since the router will mount it and will recognise it.
use `PageLayout`, take a look at my Audits PR

**Comment by waldenraines** [comment]
When I go to `/host_wizard` I see: 

```
ReactController#index is missing a template for this request format and variant. request.formats: ["text/html"] request.variant: [] NOTE! For XHR/Ajax or API requests, this action would normally respond with 204 No Content: an empty white screen. Since you're loading it in a web browser, we assume that you expected to actually render a template, not nothing, so we're showing an error to be extra-clear. If you expect 204 No Content, carry on. That's what you'll get from an XHR or API request. Give it a shot.

```

**Comment by Ron-Lavi** [comment]
@waldenraines works now when accessing the page through the url: `/host_wizard`.

**Comment by Ron-Lavi** [comment]
@glekner maybe you've faced this before, do you know why integration test are failing ?
Selection_069

**Comment by Ron-Lavi** [comment]
Added to the lab features menu :+1: 

Selection_070

**Review by waldenraines — DISMISSED** [review]
Looks good to me.  I think we can merge this pending green tests.

**Comment by Ron-Lavi** [comment]
rebased, only coverage failing but I don't think we should create tests 
as everything is going to change anyway :)

**Comment by waldenraines** [comment]
> rebased, only coverage failing but I don't think we should create tests
> as everything is going to change anyway :)

You could create some simple test for the new components that we could add to in the future.  I don't think that would be too difficult or wasteful.

File: app/controllers/react_controller.rb
Comment (ohadlevy) [inline app/controllers/react_controller.rb]: please rebase, these changes are already in core.

File: app/registries/menu/loader.rb
Comment (ohadlevy) [inline app/registries/menu/loader.rb]: you need url or url_hash, not both, in this case just remove the url_hash?

**Review by ohadlevy — CHANGES_REQUESTED** [review]

**Comment by Ron-Lavi** [comment]
rebased

**Comment by ohadlevy** [comment]
this needs a quick rebase now.. thanks.

**Comment by Ron-Lavi** [comment]
rebased

**Comment by Ron-Lavi** [comment]
Can we merge this one? @ohadlevy

**Review by ohadlevy — APPROVED** [review]

**Pull request merged**
